packages/cloudflare/src/api/cloudflare-context.ts

```diff
@@ -0,0 +1,175 @@
+import type { Context, RunningCodeOptions } from "node:vm";
+
+declare global {
+  interface CloudflareEnv {
+    NEXT_CACHE_WORKERS_KV?: KVNamespace;
+    ASSETS?: Fetcher;
+  }
+}
+
+export type CloudflareContext<
+  CfProperties extends Record<string, unknown> = IncomingRequestCfProperties,
+  Context = ExecutionContext,
+> = {
+  /**
+   * the worker's [bindings](https://developers.cloudflare.com/workers/runtime-apis/bindings/)
+   */
+  env: CloudflareEnv;
+  /**
+   * the request's [cf properties](https://developers.cloudflare.com/workers/runtime-apis/request/#the-cf-property-requestinitcfproperties)
+   */
+  cf: CfProperties | undefined;
+  /**
+   * the current [execution context](https://developers.cloudflare.com/workers/runtime-apis/context)
+   */
+  ctx: Context;
+};
+
+/**
+ * Symbol used as an index in the global scope to set and retrieve the Cloudflare context
+ *
+ * This is used both in production (in the actual built worker) and in development (`next dev`)
+ *
+ * Note: this symbol needs to be kept in sync with the one used in `src/cli/templates/worker.ts`
+ */
+const cloudflareContextSymbol = Symbol.for("__cloudflare-context__");
+
+/**
+ * Utility to get the current Cloudflare context
+ *
+ * @returns the cloudflare context
+ */
+export function getCloudflareContext<
+  CfProperties extends Record<string, unknown> = IncomingRequestCfProperties,
+  Context = ExecutionContext,
+>(): CloudflareContext<CfProperties, Context> {
+  const global = getInternalGlobalThis<CfProperties, Context>();
+
+  const cloudflareContext = global[cloudflareContextSymbol];
+
+  if (!cloudflareContext) {
+    // the cloudflare context is initialized by the worker and is always present in production/preview
+    // during local development (`next dev`) it might be missing only if the developers hasn't called
+    // the `initOpenNextCloudflareForDev` function in their Next.js config file
+    const getContextFunctionName = getCloudflareContext.name;
+    const initFunctionName = initOpenNextCloudflareForDev.name;
+    throw new Error(
+      `\n\n\`${getContextFunctionName}\` has been called during development without having used` +
```

Comment:
Could this be a single ```...string...``` to avoid `\n`, `+`?

Reply:
we could use string literals but then we'd have to de-dent them, for example using a library like [`ts-dedent`](https://www.npmjs.com/package/ts-dedent) (like wrangler does for example) or by doing that manually (like with a regex replace)

personally I prefer `\n` + `+` (no extra dependency/brittleness to deal with and it is practically as readable), but if you want to use string literals + some de-dent functionality I'm ok with that too

Reply:
You could just:

```
      throw new Error(`

\`${getContextFu...
Another line

`);
```

Reply:
oh.... I really really dislike that! 🥲

(in my opinion that sort of thing completely breaks files indentation and looks just super messy)

Reply:
keep you version then 👍